djedi/tests/test_templatetags.py

```diff
@@ -1,104 +1,117 @@
-from .compat import cmpt_context, get_template_from_string
 
 
 class TagTest(DjediTest, AssertionMixin):
-
     def render(self, source, context=None):
-        source = u'{% load djedi_tags %}' + source.strip()
-        context = cmpt_context(context or {})
-        return get_template_from_string(source).render(context).strip()
+        source = u"{% load djedi_tags %}" + source.strip()
+        return engines["django"].from_string(source).render(context).strip()
 
     def test_node_tag(self):
         html = self.render(u"{% node 'page/title' edit=False %}")
-        assert html == u''
+        assert html == u""
 
-        cio.set('i18n://sv-se@page/title.txt', u'Djedi')
-        cio.set('i18n://sv-se@page/body.txt', u'Lightning fast!')
+        cio.set("i18n://sv-se@page/title.txt", u"Djedi")
+        cio.set("i18n://sv-se@page/body.txt", u"Lightning fast!")
 
         with self.assertCache(calls=1, misses=0):
             with self.assertDB(calls=0):
-                html = self.render(u"<h1>{% node 'page/title' edit=False %}</h1><p>{% node 'page/body' edit=False %}</p>")
-                assert html == u'<h1>Djedi</h1><p>Lightning fast!</p>'
+                html = self.render(
+                    u"<h1>{% node 'page/title' edit=False %}</h1><p>{% node 'page/body' edit=False %}</p>"
+                )
+                assert html == u"<h1>Djedi</h1><p>Lightning fast!</p>"
 
         cache.clear()
 
         with self.assertCache(calls=1, misses=2):
             with self.assertDB(calls=1):
-                html = self.render(u"<h1>{% node 'page/title' edit=False %}</h1><p>{% node 'page/body' %}</p>")
-                assert html == u'<h1>Djedi</h1><p><span data-i18n="sv-se@page/body">Lightning fast!</span></p>'
+                html = self.render(
+                    u"<h1>{% node 'page/title' edit=False %}</h1><p>{% node 'page/body' %}</p>"
+                )
+                assert (
+                    html
+                    == u'<h1>Djedi</h1><p><span data-i18n="sv-se@page/body">Lightning fast!</span></p>'
+                )
 
         html = self.render(u"{% node 'foo/bar' default='bogus' %}")
         assert html == u'<span data-i18n="sv-se@foo/bar">bogus</span>'
         html = self.render(u"{% node 'l10n://foo/bar' default='bogus' %}")
         self.assertEqual(html, u'<span data-i18n="djedi@foo/bar">bogus</span>')
 
     def test_node_tag_with_default_scheme(self):
-        cio.set('i18n://sv-se@page/title.txt', u'Swedish Djedi')
+        cio.set("i18n://sv-se@page/title.txt", u"Swedish Djedi")
         html = self.render(u"{% node 'page/title' edit=False %}")
-        assert html == u'Swedish Djedi'
+        assert html == u"Swedish Djedi"
 
-        with self.settings(DJEDI={'URI_DEFAULT_SCHEME': 'l10n'}):
+        with self.settings(DJEDI={"URI_DEFAULT_SCHEME": "l10n"}):
             html = self.render(u"{% node 'page/title' edit=False %}")
-            assert html == u''
+            assert html == u""
```

Comment:
We should change all `u""` to `""` now that we've dropped py2

Reply:
We could leave it for a PR where we add pre-commit (then it's done automatically with pyupgrade, if that fits in this project)